src/PrestaShopBundle/Translation/PrestaShopTranslatorTrait.php

```diff
@@ -58,23 +58,25 @@ public function trans($id, array $parameters = [], $domain = null, $locale = nul
             $locale = null;
         }
 
-        $translated = parent::trans($id, [], $this->normalizeDomain($domain), $locale);
+        $translated = parent::trans($id, $parameters, $this->normalizeDomain($domain), $locale);
 
         // @todo to remove after the legacy translation system has ben phased out
         if ($this->shouldFallbackToLegacyModuleTranslation($id, $domain, $translated)) {
             return $this->translateUsingLegacySystem($id, $parameters, $domain, $locale);
         }
 
-        if (isset($legacy) && 'htmlspecialchars' === $legacy) {
-            $translated = call_user_func($legacy, $translated, ENT_NOQUOTES);
-        } elseif (isset($legacy)) {
-            $translated = call_user_func($legacy, $translated);
-        }
-
-        if (!empty($parameters) && $this->isSprintfString($id)) {
-            $translated = vsprintf($translated, $parameters);
-        } elseif (!empty($parameters)) {
-            $translated = strtr($translated, $parameters);
+        if (isset($legacy)) {
+            if (isset($legacy) && 'htmlspecialchars' === $legacy) {
+                $translated = call_user_func($legacy, $translated, ENT_NOQUOTES);
+            } elseif (isset($legacy)) {
+                $translated = call_user_func($legacy, $translated);
+            }
+
+            if (!empty($parameters) && $this->isSprintfString($id)) {
```

Comment:
Why do you add the `if (isset($legacy))` condition to apply those translations ?